sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -2300,6 +2300,37 @@ class Dataset[T] private[sql](
     }
   }
 
+  /**
+   * Returns a new Dataset with columns renamed.
+   * This is a no-op if schema doesn't contain existingNames in columnMap.
+   * {{{
+   *   df.withColumnRenamed(Map(
+   *     "c1" -> "first_column",
+   *     "c2" -> "second_column"
+   *   ))
+   * }}}
+   *
+   * @group untypedrel
+   * @since 2.4.0
```

Comment:
branch-2.4 is cut out. We will probably target 3.0.0 if we happen to add new APIs.